govcd/nsxt_nat_rule_test.go

```diff
@@ -0,0 +1,354 @@
+// +build network nsxt functional openapi ALL
+
+package govcd
+
+import (
+	"fmt"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	. "gopkg.in/check.v1"
+)
+
+func (vcd *TestVCD) Test_NsxtNatDnat(check *C) {
+	skipNoNsxtConfiguration(vcd, check)
+	skipOpenApiEndpointTest(vcd, check, types.OpenApiPathVersion1_0_0+types.OpenApiEndpointFirewallGroups)
+
+	org, err := vcd.client.GetOrgByName(vcd.config.VCD.Org)
+	check.Assert(err, IsNil)
+
+	nsxtVdc, err := org.GetVDCByName(vcd.config.VCD.Nsxt.Vdc, false)
+	check.Assert(err, IsNil)
+
+	edge, err := nsxtVdc.GetNsxtEdgeGatewayByName(vcd.config.VCD.Nsxt.EdgeGateway)
+	check.Assert(err, IsNil)
+
+	appPortProfiles, err := org.GetAllNsxtAppPortProfiles(nil, types.ApplicationPortProfileScopeSystem)
+	check.Assert(err, IsNil)
+
+	natRuleDefinition := &types.NsxtNatRule{
+		Name:              check.TestName() + "dnat",
+		Description:       "description",
+		Enabled:           true,
+		RuleType:          types.NsxtNatRuleTypeDnat,
+		ExternalAddresses: edge.EdgeGateway.EdgeGatewayUplinks[0].Subnets.Values[0].PrimaryIP,
```

Comment:
Is there any chance that a component of the path down to `PrimaryIP` could be `nil`?

Reply:
It should not.
What happens is that Edge Gateway requires IP addresses (at least 1) from external network. It does it set as `primary_ip` by default.


Here is a screenshot from fresh created NSX-T Edge Gateway

<img width="1134" alt="image" src="https://user-images.githubusercontent.com/15804230/123106887-b768e600-d441-11eb-9823-0290c17cd2e4.png">

Now I have added additional explicit safeguard (validation) that it is not empty. I have never seen it, but at least tests should warn early if that changes in future.